pkg/ring/tokens.go

```diff
@@ -0,0 +1,102 @@
+package ring
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+)
+
+const (
+	// TokensVersion1 is the version is a simple list of tokens.
+	TokensVersion1 = 1
+)
+
+// Tokens is a simple list of tokens.
+type Tokens []uint32
+
+// StoreToFile stores the tokens in the given directory.
+func (t Tokens) StoreToFile(tokenFilePath string) (e error) {
+	if tokenFilePath == "" {
+		return errors.New("path is empty")
+	}
+	tmp := tokenFilePath + ".tmp"
+	f, err := os.OpenFile(tmp, os.O_WRONLY|os.O_CREATE, 0666)
+	if err != nil {
+		return err
+	}
+
+	fileClosed := false
+	defer func() {
+		if !fileClosed {
+			if err := f.Close(); e == nil && err != nil {
+				e = err
+			}
+		}
+		// RemoveAll returns no error when tmp doesn't exist so it is safe to always run it.
+		if err := os.RemoveAll(tmp); err != nil {
+			level.Error(util.Logger).Log("msg", "error deleting temporary file", "err", err)
```

Comment:
This looks more a `level.Warn` to me, because the even if this error occurs the application is not negatively affected (we just miss the opportunity to cleanup a little file from disk).